### PR TITLE
Issue 3268: Standalone mode security documentation

### LINKED ISSUE #3268: Improve documentation about setting up TLS for standalone mode
**Problem description**
The documentation doesn't explain well how to configure SSL/TLS in Pravega standalone mode for SSL/TLS communications. As a result, it's difficult to set it up and get it up and running quickly for someone who is not already familiar with the process. 

Also, it needs to be updated to explain additional configuration items pertaining to keystores and truststores which are used internally for inter-component communications and the REST server. 

**Problem location**
Documentation

**Suggestions for an improvement**
Documentation about how to run Pravega standalone mode with TLS needs the following enhancements and updates: 
* Explain how to use provided PKI material such as certificates, keys, keystores and truststores
* Update the descriptions for configuration parameters to explain them better
* Add descriptions about configurations required for configuring REST server with TLS, which are missing from the documentation

### PR DESCRIPTION
**Pull request by ravisharda**
**Change log description**  
Adds detailed information about configuring security for Pravega standalone. 

**Purpose of the change**  
Resolves #3268.

**What the code does**  
Adds detailed information about configuring security. Updates existing documentation to refactor the documents. 

**How to verify it**  
N/A

### CODE REVIEW

File: documentation/src/docs/deployment/run-local.md [hunk 1]
```diff
@@ -7,25 +7,30 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 -->
-# Running Pravega
+# Running Pravega in Local Machine
 
+As an alternative to running Pravega on a cluster of machines, you may run Pravega on a local machine. Running Pravega locally on a single host, allows you to get started with Pravega quickly. Running Pravega locally is especially suitable for development and testing purposes.
```
Comment (andreipaduroiu) [inline documentation/src/docs/deployment/run-local.md]: remove comma between `host` and `allows`

File: documentation/src/docs/deployment/run-local.md [hunk 2]
```diff
@@ -7,25 +7,30 @@ You may obtain a copy of the License at
...
 
-Running in local machine allows us to get started using Pravega very quickly. Running Pravega in Standalone mode is suitable for development and testing Pravega applications.
-
-The prerequisites for running in local machine is described below.
+You may run Pravega in local machine either in either standalone mode or in distributed mode using Docker Compose.
```
Comment (andreipaduroiu) [inline documentation/src/docs/deployment/run-local.md]: `in local machine` -> `on your local machine`

File: documentation/src/docs/deployment/run-local.md [hunk 3]
```diff
@@ -7,25 +7,30 @@ You may obtain a copy of the License at
...
 
 ## Standalone Mode
 
+In standalone mode, the Pravega cluster is accessible from clients through the `localhost` interface only. Controller REST API's, however, are accessible from remote hosts/machines.
```
Comment (andreipaduroiu) [inline documentation/src/docs/deployment/run-local.md]: `API's` -> `APIs`

File: documentation/src/docs/security/securing-standalone-mode-cluster.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+-->
+
+# Setting Up Security for a Standalone Mode Cluster
+
+Security in Pravega is off by default. You may start Pravega Standalone mode with security enabled, by modifying the security configuration before launching it.
```
Comment (andreipaduroiu) [inline documentation/src/docs/security/securing-standalone-mode-cluster.md]: remove comma before `by`

File: documentation/src/docs/security/securing-standalone-mode-cluster.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+Depending on how you configure security, either or both of the following occurs:
+
+1. Client-server and internal communications are encrypted using SSL/TLS
+2. Server authenticates and authorizes client requests
+
+For standalone mode clusters, you may enable SSL/TLS, and/ `auth` (short for Authentication and Authorization). We recommend that you enable both. The subsections below describe how to enable them.
+
+## Enabling SSL/TLS and Auth
+
+The configuration parameter `singlenode.enableTls` determines whether SSL/TLS is enabled in a standalone mode cluster. It's default value is `false`, and therefore, SSL/TLS is disabled by default.
```
Comment (andreipaduroiu) [inline documentation/src/docs/security/securing-standalone-mode-cluster.md]: `It's` -> `Its`

File: documentation/src/docs/security/securing-standalone-mode-cluster.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+     a) Create a file containing 1 or more entries of the form  `<user>:<password>:<acl>;`.
+
+        For example: `jdoe:strong-password:*,READ_UPDATE;`
+
+     b) Use the PasswordCreatorTool to create a new file with the passwords encrypted. Configure this new file instead of the supplied `passwd` file.
+
+4. If enabling SSL/TLS, ensure that the server's certificate is trusted.
+
+   Note: This step is only needed the first time you run SSL/TLS-enabled standalone mode with a certificate.If you have already imported the certificate earlier, you don't need to do it again the next time you launch the server.
```
Comment (andreipaduroiu) [inline documentation/src/docs/security/securing-standalone-mode-cluster.md]: Add a space between sentences (`certificate. If`)

File: documentation/src/docs/deployment/run-local.md (hunk 1 above)
Comment (ravisharda) [reply documentation/src/docs/deployment/run-local.md]: Done.

File: documentation/src/docs/deployment/run-local.md (hunk 3 above)
Comment (ravisharda) [reply documentation/src/docs/deployment/run-local.md]: Done.

File: documentation/src/docs/security/securing-standalone-mode-cluster.md (hunk 4 above)
Comment (ravisharda) [reply documentation/src/docs/security/securing-standalone-mode-cluster.md]: Done.

File: documentation/src/docs/security/securing-standalone-mode-cluster.md (hunk 6 above)
Comment (ravisharda) [reply documentation/src/docs/security/securing-standalone-mode-cluster.md]: Done. You sure have a keen eye. :)

File: documentation/src/docs/security/securing-standalone-mode-cluster.md (hunk 5 above)
Comment (ravisharda) [reply documentation/src/docs/security/securing-standalone-mode-cluster.md]: Fixed.

**Pull request merged**